src/js/common/schemaform/reducers/index.js

```diff
@@ -17,16 +17,30 @@ import { SET_DATA,
 } from '../actions';
 
 function recalculateSchemaAndData(initialState) {
-  return Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
+  const retVal = Object.keys(_.omit(['privacyAgreementAccepted', 'submission'], initialState))
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = Object.keys(state).reduce((carry, pageName) => {
```

Comment:
This actually flattens it properly; with `_.mapValues()`, it was still broken up by page. Woops. Fortunately, this also clocks in either faster than `_.mapValues()` or insignificantly slower. Either way, it was `console.time()`ing on my machine at ~0.1ms, so it's pretty insignificant.